### ISSUE TITLE
fromDirectGeometry() seems to imply 3 indices per geometry

### ISSUE THREAD

**Issue by nraynaud** [mrdoob/three.js#7254]
I am trying to debug some bufferedGeometry LineSegments code with the dev branch, and I'm getting some "GL_INVALID_OPERATION : glDrawElements: range out of bounds for buffer" errors (it was working fine with an older three.js, and a JS check of the index attribute against the position attribute doesn't bring anything suspicious, I think the buffers are fine, but there might be a kerfuffle  around the upload to GL).

During my investigations, I found some suspicious code:
I think that https://github.com/mrdoob/three.js/blob/master/src/core/BufferGeometry.js#L509 implies 3 indices per geometry which is fine for triangle, but not for lines and points.

To be honest, I don't really understand this code and it's not yet documented, I hope I'm not making people lose their time.

I am also sensing some headaches around the normals computation code, if the BufferGeometry doesn't know if it is encoding a triangle soup, and line soup or a point soup, somebody might be having a migraine, what about moving all that to the Mesh class?

Sorry if that's a false alarm, rest assured that I also spend a *lot* of time trying to catch up with the three.js codebase (and the weirdo world of webgl extensions in general), I'm not simply dumping my work on you.

**Comment by mrdoob** [comment]
Could you share the code that doesn't work anymore?

`.fromDirectGeometry()` shouldn't be executed for Lines nor Points.

**Comment by nraynaud** [comment]
Sorry it's a huge non-isolated pile of code, I just bypassed it, and I'll move on to something else. And while googling for this problem I actually found that I should have used another part of the API (the drawcalls()).

here is the removal commit, as the head of the trail:
https://github.com/nraynaud/webgcode/commit/3f8463af8c4a7cc87fbef37d9d728caf6656f7c0

when increasing the size of an indexed bufferedGeometry representing a polyline, the error would pop. When checking in JS, all the indices in the index buffer were smaller than the position buffer length (ie I had no overflow), as if the new index buffer was uploaded and the object rendered before the new position buffer was uploaded.

You are welcome to close if you don't want to dig into my code.
